dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -451,6 +452,20 @@ public ApiToken findApiTokenByUser(AuthenticatedUser au) {
         }
     }
 
+    public List<ApiToken> findAllApiTokensByUser(AuthenticatedUser user) {
```

Comment:
IMO instead of having to juggle with api keys we could enforce that single user can have only one api key. That way we have a little simpler situation. Maybe we could just add unique constraint in ApiToken for authenticated user column? What do you think?

Reply:
The idea is good, but I think it should be done in a new task, as it needs some further analysis and refactor.